### PR TITLE
Add system to namespace filter

### PR DESCRIPTION
**Pull request by dngroth**

### CODE REVIEW

File: Test.Utility/ContainerInterfaceRetriever.cs [hunk 1]
```diff
@@ -19,10 +19,15 @@ public static IEnumerable<Type> GetAssemblyInterfaces(IContainer container)
...
+                .Where(IsValidType);
 
-        private static bool BelongsToAutofacNamespace(Type type)
-            => NamespaceContains(type, nameof(Autofac));
+        private static bool IsValidType(Type type)
+            => type.IsInterface &&
+               !type.IsGenericType &&
+               !type.BelongsToNamespaces(new[] { nameof(Autofac), nameof(System) });
+
+        private static bool BelongsToNamespaces(this Type type, IEnumerable<string> namespaces)
```
Comment (jnferner) [inline Test.Utility/ContainerInterfaceRetriever.cs]: Slightly inaccurate. How about `BelongsToAnyNamespace`?

File: Test.Utility/ContainerInterfaceRetriever.cs [hunk 2]
```diff
@@ -19,10 +19,15 @@ public static IEnumerable<Type> GetAssemblyInterfaces(IContainer container)
...
                 .GetTypes()
-                .Where(type => type.IsInterface && !type.IsGenericType && !BelongsToAutofacNamespace(type));
+                .Where(IsValidType);
 
-        private static bool BelongsToAutofacNamespace(Type type)
-            => NamespaceContains(type, nameof(Autofac));
+        private static bool IsValidType(Type type)
+            => type.IsInterface &&
+               !type.IsGenericType &&
+               !type.BelongsToNamespaces(new[] { nameof(Autofac), nameof(System) });
```
Comment (jnferner) [inline Test.Utility/ContainerInterfaceRetriever.cs]: Move into constant

Comment (dngroth) [reply Test.Utility/ContainerInterfaceRetriever.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/24/commits/cf32104e6a7ed1adbca3ccdf08097e7c1c0098f4

File: Test.Utility/ContainerInterfaceRetriever.cs (hunk 1 above)
Comment (dngroth) [reply Test.Utility/ContainerInterfaceRetriever.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/24/commits/cf32104e6a7ed1adbca3ccdf08097e7c1c0098f4

File: Test.Utility/ContainerInterfaceRetriever.cs (hunk 2 above)
Comment (jnferner) [reply Test.Utility/ContainerInterfaceRetriever.cs]: `Types`*

File: Test.Utility.Test/Stubs/ITestInterface.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+namespace Messerli.Test.Utility.Test.Stubs
+{
+    public interface ITestInterface
+    {
+    }
+
+    public interface ITestInterface2
+    {
+    }
+}
```
Comment (jnferner) [inline Test.Utility.Test/Stubs/ITestInterface.cs]: Add terminating newline.
There seems to be no .editorconfig

File: Test.Utility/ContainerInterfaceRetriever.cs [hunk 4]
```diff
@@ -8,6 +8,9 @@ namespace Messerli.Test.Utility
 {
     public static class ContainerInterfaceRetriever
     {
+        private static readonly IEnumerable<string> ExcludedNamespaces =
+            new[] { nameof(System), "Foundation", "Internal", nameof(Autofac) };
```
Comment (jnferner) [inline Test.Utility/ContainerInterfaceRetriever.cs]: Can't you use `nameof` here?

Comment (dngroth) [reply Test.Utility/ContainerInterfaceRetriever.cs]: no

File: Test.Utility.Test/Stubs/ITestInterface.cs (hunk 3 above)
Comment (dngroth) [reply Test.Utility.Test/Stubs/ITestInterface.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/24/commits/c8be4571c294ab68dabf6b628a4ae6b21c0d5eda

**Pull request merged**